### PR TITLE
Bind DVC data registry

### PR DESCRIPTION
**Pull request by AlexVCaron**
# Quick description

Everything required in Scilpy to throw away gdrive and embrace DVC !

From discussions :

- Scil_data : contains all datasets we need to test, validate, try code, and more (in ./data). Contains packages of datasets easy to download and version in one line (in ./store).
- Scilpy : has a DVC handle pointing to Scil_data. Loads files and packages by interfacing through DVC.
- Other repos : use the same gammick as Scilpy to load datasets they require.

Looking at everything (go through [scil_data](https://github.com/AlexVCaron/scil_data), it seems like a lot of duplication, disk space must be wasted !!!!! Not at all.

Everything contained in the repository is index files for DVC, kilobyte sized, super light for Git. The data *per say* is all contained [here](https://scil.usherbrooke.ca/scil_test_data/dvc-store). You won't be able to recognize anything, that's the point. Every file indexed by DVC gets hashed and the md5 is used then to create the filepath.

Doing so, we can allow duplication in the Git repository - since md5 always match, even when the names and timestamps differ - and there will never be duplication on the data server.

Using DVC is very similar to how we used the fetcher, with the addition that we now need to keep track of test packages revisions. To create a new test data package :

1. Clone [scil_data](https://github.com/AlexVCaron/scil_data) and follow the documentation to create a new package.
2. Once uploaded, copy the commit at which the package was published and add it to `data/test_descriptors.yml` under your test case name.
3. Use the `pull_test_case_package` function from `scilpy.io.dvc` in your test script to make the data available when testing.

...

## Type of change

Check the relevant options.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Provide data, screenshots, command line to test (if relevant)

The aodf_metrics test case is implemented as POC :

pytest scripts/tests/test_aodf_metrics.py

# Checklist

- [x] My code follows the style guidelines of this project (run [autopep8](https://pypi.org/project/autopep8/))
- [x] I added relevant citations to scripts, modules and functions docstrings and descriptions
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I moved all functions from the script file (except the argparser and main) to scilpy modules
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-880/1/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-880/2/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-880/3/

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

File: .dvc/config [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[cache]
+    dir = /home/scil/.scilpy/dvc-cache/
```
Comment (arnaudbore) [inline .dvc/config:2]: absolut path ?

File: scilpy/io/dvc.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+
+from dvc import config, api
+import os
+import yaml
+
+from scilpy import SCILPY_HOME, SCILPY_ROOT
+
+
+DVC_REPOSITORY = "https://github.com/AlexVCaron/scil_data.git"
```
Comment (arnaudbore) [inline scilpy/io/dvc.py:11]: we should move this repo before merging

File: scilpy/__init__.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+
+
+def get_home():
```
Comment (arnaudbore) [inline scilpy/__init__.py:3]: Not sure about this. I don't think this should go there.

**Review by arnaudbore — COMMENTED** [review]
Small comments, I would like @EmmaRenauld or/and @karanphil to give their feedback about the __init__.py first before moving forward.

File: scilpy/__init__.py (hunk 3 above)
Comment (arnaudbore) [reply scilpy/__init__.py:3]: We already have a get_home function.. let's talk about this.

File: .dvc/config (hunk 1 above)
Comment (AlexVCaron) [reply .dvc/config:2]: I will move the whole cache field inside of io/dvc, create it dynamically in python. The thing is this file doesn't expand environment variables, so I cannot assign $HOME and make the behavior similar to before.

File: scilpy/io/dvc.py (hunk 2 above)
Comment (AlexVCaron) [reply scilpy/io/dvc.py:11]: Doing it live this afternoon :wink:

Comment (AlexVCaron) [reply scilpy/io/dvc.py:11]: Doing it live tomorrow !

**Comment by AlexVCaron** [comment]
I found a bug in `DVC` (in a sub dependency `dvc-data`). It makes the cache slightly useless for now, but doesn't undermine this PR. Still we can follow iterative/dvc-data#512

**Review by arnaudbore — APPROVED** [review]
GTG

**Pull request merged**